### ISSUE TITLE
Intermittent scrobble failures

### ISSUE THREAD

**Issue by SenorSmartyPants** [jellyfin/jellyfin-plugin-trakt#67]
JF 10.7.0 
plugin 11

Sometimes plugin will no scrobble to trakt. I notice this after the fact. I'll look at my previous night of watching tv episodes and maybe half were scrobbled to trakt.

**Comment by lmaonator** [comment]
I think this could be related to Trakts switch from TVDB as primary source to TMDB.
I noticed episodes that fail to scrobble don't have a TVDB link on trakt in the bottom left.

Can you see if that's the same for you?

I think considering Jellyfin also dropped TVDB as primary metadata provider, just like Trakt, this plugin should scrobble using TMDB as well.

**Comment by SenorSmartyPants** [comment]
I think you're on to something. This is the one episode that didn't scrobble tonight. No tvdb I'd on trakt. 

https://trakt.tv/shows/last-week-tonight-with-john-oliver/seasons/8/episodes/5

I think you can send multiple IDs when scrobbling to trakt. Just send all the ones JF has, season episode numbers too if possible. From my experimenting with the trakt API years ago they do a good job of picking the most specific IDs.

**Comment by SenorSmartyPants** [comment]
Looking closer at this episode, my JF only has a TVDB id for it, no IMDB, or TMDB ids. Trakt only has TMDB and IMDB. So no overlap.

I have TVDB plugin scraper and TMDB scraper enabled for my TV library.

**Comment by SenorSmartyPants** [comment]
I did a little digging in the code. Here seems to be the problem area.

https://github.com/jellyfin/jellyfin-plugin-trakt/blob/8621aa3a34a347a37f2e437cb3a44bc0acd7635d/Trakt/Api/TraktApi.cs#L171-L186

If episode has TVDB, IndexNumber and no IndexNumberEnd, then only TVDBid is passed to Trakt in the scrobble. But **_Trakt is NOT guaranteed_** to have a TVDBid anymore. So then no matching happens. This is very much related to Trakt switching away from TVDB.

I propose sending all episode providerIDs present on an episode, including season and episode numbers. As well as show ID information. As done in this code block.

https://github.com/jellyfin/jellyfin-plugin-trakt/blob/8621aa3a34a347a37f2e437cb3a44bc0acd7635d/Trakt/Api/TraktApi.cs#L194-L215

**Comment by SenorSmartyPants** [comment]
That Last Week Tonight episode now has TVDBid info. But this one currently does not. So it is a recurring problem.
https://trakt.tv/shows/mayans-m-c/seasons/3/episodes/6

**Comment by jackmeat** [comment]
This issue is driving me nuts and I assume you as well judging by how long ago you noticed. I just noticed trakt wasn't picking up me watching Saturday night live so I knew scribbling wasn't working. Assuming the popularity of the show i checked and there was an IMDB ID for it which was NOT populated in Jellyfin. I figured inputting the ID into the metadata would solve this issue for just this one episode, but that did NOT even work.
